lib/zendesk_apps_support/package.rb

```diff
@@ -101,14 +101,22 @@ def readified_js(app_name, app_id, asset_url_prefix, settings={})
       app_class_name = "app-#{app_id}"
       author = manifest[:author]
       framework_version = manifest[:frameworkVersion]
-      templates = manifest[:noTemplate] ? {} : compiled_templates(app_id, asset_url_prefix)
+      single_install = manifest[:singleInstall] || false
+      no_template = manifest[:noTemplate]
+      templates = no_template ? {} : compiled_templates(app_id, asset_url_prefix)
 
       settings["title"] = name
 
+      app_settings = {
+        :location => location,
+        :noTemplate => no_template,
+        :singleInstall => single_install
+      }.keep_if { |k,v| !v.nil? }
```

Comment:
Sure, I'll use `select`.  `keep_if` just seemed novel, and semantically
meaningful in my use case 😸

On Wednesday, January 28, 2015, Po Chen <notifications@github.com> wrote:

> In lib/zendesk_apps_support/package.rb
> <https://github.com/zendesk/zendesk_apps_support/pull/77#discussion_r23659664>
> :
>
> >
> >        settings["title"] = name
> >
> > +      app_settings = {
> > +        :location => location,
> > +        :noTemplate => no_template,
> > +        :singleInstall => single_install
> > +      }.keep_if { |k,v| !v.nil? }
>
> or even .reject { |k, v| v.nil? }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/zendesk_apps_support/pull/77/files#r23659664>.
>


-- 
Stefan Vizzari
Snr. Software Engineer - APAC

Zendesk <http://www.zendesk.com/>
Twitter: @s_vizzari  |  Skype: zendesk.svizzari | Phone: +61 407 511 960